lib/model/services/ble/ble_service.dart

```diff
@@ -11,8 +10,8 @@ import 'package:flutter_ble_lib/flutter_ble_lib.dart';
 class BLEService extends ChangeNotifier {
   static BleManager bleManager = BleManager();
 
-  List<Peripheral> _devicesList = new List<Peripheral>();
-  PermissionStatus _locationPermissionStatus;
+  final List<Peripheral> _devicesList = <Peripheral>[];
```

Comment:
The device list should not be final. I think it is fine for the moment as it needs to be rewritten anyways to give a more clean abstraction

Reply:
Just to be clear I think this just says the parent reference is final, not the contents of the device list. (at least that's my understanding) I think the linter picked this up as never being assigned to so recommended the final but happy to remove if plans are otherwise. (just the linter will fail then)